File: api/src/models/user-model.ts
import { Document, Schema, model } from 'mongoose';

export interface IUser extends Document {
  name: string;
  email: string;
  password: string;
  token: string;
}

const userSchema = new Schema({
  name: {
    type: String,
    required: true
  },
  email: { 
    type: String,
    required: true
  },
  password: {
    type: String,
    required: true
  },
  token: {
    type: String,
    default: null
  }
});

export const UserModel = model<IUser>('Users', userSchema);

File: app/lib/models/user_model.dart
class UserModel {
  final String name;
  final String email;
  final String password;
  final String token;
  final String id;

  UserModel({
    required this.name,
    required this.email,
    required this.password,
    required this.token,
    required this.id,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      name: json['name'],
      email: json['email'],
      password: json['password'],
      token: json['token'],
      id: json['_id'],
    );
  }
}
